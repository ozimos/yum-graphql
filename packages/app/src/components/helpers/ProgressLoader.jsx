import React from 'react';
import PropTypes from 'prop-types';

import LinearProgress from '@material-ui/core/LinearProgress';

const ProgressLoader = props =>
  (
    <div className="progression">
      <LinearProgress
        style={{ height: '10px' }}
        variant="determinate"
        value={props.upload}
      />
    </div>
  );

ProgressLoader.propTypes = {
  upload: PropTypes.number.isRequired,
};

export default ProgressLoader;

