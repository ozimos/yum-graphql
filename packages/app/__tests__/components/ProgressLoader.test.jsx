/* global React:false, shallow:false toJson:false */

import ProgressLoader from '../../src/components/helpers/ProgressLoader';

const props = {
  upload: 10
};
describe('Progress Loader', () => {

  const setup = () => shallow(<ProgressLoader {...props} />);

  it('renders correctly', () => {
    const wrapper = setup();
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
