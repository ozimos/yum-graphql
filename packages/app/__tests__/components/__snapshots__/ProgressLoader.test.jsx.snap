// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Progress Loader renders correctly 1`] = `
<div
  className="progression"
>
  <WithStyles(LinearProgress)
    style={
      Object {
        "height": "10px",
      }
    }
    value={10}
    variant="determinate"
  />
</div>
`;
